### DISCUSSION TITLE
how to configuration rsa_private_key   on .env file ?

### DISCUSSION THREAD

**Discussion by Batgerel53** [saleor/saleor#11269] [Q&A]
django.core.exceptions.ImproperlyConfigured: Unable to load provided PEM private key. ('Could not deserialize key data. The data may be in an incorrect format, it may be encrypted with an unsupported algorithm, or it may be an unsupported key type (e.g. EC curves with explicit parameters).', [_OpenSSLErrorWithText(code=151584876, lib=9, reason=108, reason_text=b'error:0909006C:PEM routines:get_name:no start line')])

**Comment by peelar** [comment]
Hello @Batgerel53 👋 

Can you please describe what you do, step by step, and provide formatted logs and errors?

**Reply by Batgerel53** [reply]
openssl genrsa -out rsa.private 1024
I tried to create it like this.

**Reply by Batgerel53** [reply]
![dj](https://user-images.githubusercontent.com/56902009/204258837-49b276fd-4cd1-4b02-bbb9-5fa78f59bcdd.PNG)

**Reply by Batgerel53** [reply]
django.core.exceptions.ImproperlyConfigured: Unable to load provided PEM private key. ('Could not deserialize key data. The data may be in an incorrect format, it                                                                            may be encrypted with an unsupported algorithm, or it may be an unsupported key type (e.g. EC curves with explicit parameters).', [_OpenSSLErrorWithText(code=503                                                                           841036, lib=60, reason=524556, reason_text=b'error:1E08010C:DECODER routines::unsupported')])

**Reply by Batgerel53** [reply]
@karolkielecki  @peelar  help me please

**Reply by Batgerel53** [reply]
saleor version 3.5.1

**Comment by akhial** [comment]
Try replacing the newlines in your key with '\n', and make sure the key is correct.
Something like:
```
-----BEGIN PRIVATE KEY-----
...
-----END PRIVATE KEY-----
```

**Reply by Batgerel53** [reply]
@akhial  Thank you for helping me. no difference

**Comment by korycins** [comment]
@Batgerel53 
```
ssh-keygen -m PEM -t rsa -b 2048 -f private.pem
export RSA_PRIVATE_KEY=`cat private.pem` << this is only to show that you need to copy your private.pem key to Saleor's variable>>
export RSA_PRIVATE_PASSWORD="<<the password to pem file>>"
```
or
```
openssl genrsa -out private-key.pem 2048
export RSA_PRIVATE_KEY=`cat private-key.pem` << this is only to show that you need to copy your private.pem key to Saleor's variable>>
```

**Reply by Batgerel53** [reply]
@korycins  Hello.
I get the same error after doing this step and adding it to the .env variable.

**Comment by Batgerel53** [comment]
My project is hosted on apache server.  Please explain in more detail. How to include in .env file? Is it correct to give the key to the .env variable?
